Build failure: python313Packages.pympler

## Steps To Reproduce

Steps to reproduce the behavior:

`nix build nixpkgs#python313Packages.pympler`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/cgvmjf4sc0j8d59dlzygm6pcaaqc6l1l-pympler-1.1.tar.gz
source root is pympler-1.1
setting SOURCE_DATE_EPOCH to timestamp 1719603889 of file pympler-1.1/setup.cfg
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
running egg_info
writing Pympler.egg-info/PKG-INFO
writing dependency_links to Pympler.egg-info/dependency_links.txt
writing requirements to Pympler.egg-info/requires.txt
writing top-level names to Pympler.egg-info/top_level.txt
reading manifest file 'Pympler.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
no previously-included directories found matching 'doc/source'
no previously-included directories found matching 'doc/build'
adding license file 'LICENSE'
adding license file 'NOTICE'
writing manifest file 'Pympler.egg-info/SOURCES.txt'
* Building wheel...
running bdist_wheel
running build
running build_py
creating build/lib/pympler
copying pympler/__init__.py -> build/lib/pympler
copying pympler/asizeof.py -> build/lib/pympler
copying pympler/charts.py -> build/lib/pympler
copying pympler/classtracker.py -> build/lib/pympler
copying pympler/classtracker_stats.py -> build/lib/pympler
copying pympler/garbagegraph.py -> build/lib/pympler
copying pympler/mprofile.py -> build/lib/pympler
copying pympler/muppy.py -> build/lib/pympler
copying pympler/panels.py -> build/lib/pympler
copying pympler/process.py -> build/lib/pympler
copying pympler/refbrowser.py -> build/lib/pympler
copying pympler/refgraph.py -> build/lib/pympler
copying pympler/summary.py -> build/lib/pympler
copying pympler/tracker.py -> build/lib/pympler
copying pympler/web.py -> build/lib/pympler
creating build/lib/pympler/util
copying pympler/util/__init__.py -> build/lib/pympler/util
copying pympler/util/bottle.py -> build/lib/pympler/util
copying pympler/util/compat.py -> build/lib/pympler/util
copying pympler/util/stringutils.py -> build/lib/pympler/util
copying pympler/py.typed -> build/lib/pympler
creating build/lib/pympler/templates
copying pympler/templates/memory_panel.html -> build/lib/pympler/templates
copying pympler/templates/asized_referents.tpl -> build/lib/pympler/templates
copying pympler/templates/footer.tpl -> build/lib/pympler/templates
copying pympler/templates/garbage.tpl -> build/lib/pympler/templates
copying pympler/templates/garbage_index.tpl -> build/lib/pympler/templates
copying pympler/templates/header.tpl -> build/lib/pympler/templates
copying pympler/templates/index.tpl -> build/lib/pympler/templates
copying pympler/templates/process.tpl -> build/lib/pympler/templates
copying pympler/templates/referents.tpl -> build/lib/pympler/templates
copying pympler/templates/stacktrace.tpl -> build/lib/pympler/templates
copying pympler/templates/tracker.tpl -> build/lib/pympler/templates
copying pympler/templates/tracker_class.tpl -> build/lib/pympler/templates
copying pympler/templates/jquery.flot.min.js -> build/lib/pympler/templates
copying pympler/templates/jquery.flot.stack.min.js -> build/lib/pympler/templates
copying pympler/templates/jquery.flot.tooltip.min.js -> build/lib/pympler/templates
copying pympler/templates/style.css -> build/lib/pympler/templates
creating build/lib/pympler/static
copying pympler/static/jquery.sparkline.min.js -> build/lib/pympler/static
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/pympler
copying build/lib/pympler/__init__.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/asizeof.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/charts.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/classtracker.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/classtracker_stats.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/garbagegraph.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/mprofile.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/muppy.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/panels.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/process.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/refbrowser.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/refgraph.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/summary.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/tracker.py -> build/bdist.linux-x86_64/wheel/./pympler
copying build/lib/pympler/web.py -> build/bdist.linux-x86_64/wheel/./pympler
creating build/bdist.linux-x86_64/wheel/pympler/util
copying build/lib/pympler/util/__init__.py -> build/bdist.linux-x86_64/wheel/./pympler/util
copying build/lib/pympler/util/bottle.py -> build/bdist.linux-x86_64/wheel/./pympler/util
copying build/lib/pympler/util/compat.py -> build/bdist.linux-x86_64/wheel/./pympler/util
copying build/lib/pympler/util/stringutils.py -> build/bdist.linux-x86_64/wheel/./pympler/util
copying build/lib/pympler/py.typed -> build/bdist.linux-x86_64/wheel/./pympler
creating build/bdist.linux-x86_64/wheel/pympler/templates
copying build/lib/pympler/templates/memory_panel.html -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/asized_referents.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/footer.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/garbage.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/garbage_index.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/header.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/index.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/process.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/referents.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/stacktrace.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/tracker.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/tracker_class.tpl -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/jquery.flot.min.js -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/jquery.flot.stack.min.js -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/jquery.flot.tooltip.min.js -> build/bdist.linux-x86_64/wheel/./pympler/templates
copying build/lib/pympler/templates/style.css -> build/bdist.linux-x86_64/wheel/./pympler/templates
creating build/bdist.linux-x86_64/wheel/pympler/static
copying build/lib/pympler/static/jquery.sparkline.min.js -> build/bdist.linux-x86_64/wheel/./pympler/static
running install_egg_info
running egg_info
writing Pympler.egg-info/PKG-INFO
writing dependency_links to Pympler.egg-info/dependency_links.txt
writing requirements to Pympler.egg-info/requires.txt
writing top-level names to Pympler.egg-info/top_level.txt
reading manifest file 'Pympler.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
no previously-included directories found matching 'doc/source'
no previously-included directories found matching 'doc/build'
adding license file 'LICENSE'
adding license file 'NOTICE'
writing manifest file 'Pympler.egg-info/SOURCES.txt'
Copying Pympler.egg-info to build/bdist.linux-x86_64/wheel/./Pympler-1.1-py3.13.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/Pympler-1.1.dist-info/WHEEL
creating '/build/pympler-1.1/dist/.tmp-drxivcjp/Pympler-1.1-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'pympler/__init__.py'
adding 'pympler/asizeof.py'
adding 'pympler/charts.py'
adding 'pympler/classtracker.py'
adding 'pympler/classtracker_stats.py'
adding 'pympler/garbagegraph.py'
adding 'pympler/mprofile.py'
adding 'pympler/muppy.py'
adding 'pympler/panels.py'
adding 'pympler/process.py'
adding 'pympler/py.typed'
adding 'pympler/refbrowser.py'
adding 'pympler/refgraph.py'
adding 'pympler/summary.py'
adding 'pympler/tracker.py'
adding 'pympler/web.py'
adding 'pympler/static/jquery.sparkline.min.js'
adding 'pympler/templates/asized_referents.tpl'
adding 'pympler/templates/footer.tpl'
adding 'pympler/templates/garbage.tpl'
adding 'pympler/templates/garbage_index.tpl'
adding 'pympler/templates/header.tpl'
adding 'pympler/templates/index.tpl'
adding 'pympler/templates/jquery.flot.min.js'
adding 'pympler/templates/jquery.flot.stack.min.js'
adding 'pympler/templates/jquery.flot.tooltip.min.js'
adding 'pympler/templates/memory_panel.html'
adding 'pympler/templates/process.tpl'
adding 'pympler/templates/referents.tpl'
adding 'pympler/templates/stacktrace.tpl'
adding 'pympler/templates/style.css'
adding 'pympler/templates/tracker.tpl'
adding 'pympler/templates/tracker_class.tpl'
adding 'pympler/util/__init__.py'
adding 'pympler/util/bottle.py'
adding 'pympler/util/compat.py'
adding 'pympler/util/stringutils.py'
adding 'Pympler-1.1.dist-info/LICENSE'
adding 'Pympler-1.1.dist-info/METADATA'
adding 'Pympler-1.1.dist-info/NOTICE'
adding 'Pympler-1.1.dist-info/WHEEL'
adding 'Pympler-1.1.dist-info/top_level.txt'
adding 'Pympler-1.1.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Successfully built Pympler-1.1-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for Pympler-1.1-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed Pympler-1.1-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/7j4aa3psxzx5pbxkk3419s7riq06350s-python3.13-pympler-1.1
checking for references to /build/ in /nix/store/7j4aa3psxzx5pbxkk3419s7riq06350s-python3.13-pympler-1.1...
patching script interpreter paths in /nix/store/7j4aa3psxzx5pbxkk3419s7riq06350s-python3.13-pympler-1.1
stripping (with command strip and flags -S -p) in  /nix/store/7j4aa3psxzx5pbxkk3419s7riq06350s-python3.13-pympler-1.1/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/gnvdjncpfd1wpmgn9hb42rpxydwxh0n8-python3.13-pympler-1.1-dist
checking for references to /build/ in /nix/store/gnvdjncpfd1wpmgn9hb42rpxydwxh0n8-python3.13-pympler-1.1-dist...
patching script interpreter paths in /nix/store/gnvdjncpfd1wpmgn9hb42rpxydwxh0n8-python3.13-pympler-1.1-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.13.1, pytest-8.3.3, pluggy-1.5.0
rootdir: /build/pympler-1.1
configfile: pyproject.toml
collecting ... 
collected 127 items / 9 deselected / 118 selected                              

test/asizeof/test_asizeof.py .........................................   [ 34%]
test/gui/test_garbage.py FF...F..                                        [ 41%]
test/gui/test_web.py 127.0.0.1 - - [04/Jan/2025 02:44:39] "GET /garbage/graph/0?reduce=on HTTP/1.1" 200 0
....Bottle v0.13.1 server starting up (using PymplerServer())...
Listening on http://localhost:64546/
Hit Ctrl-C to quit.

..                                              [ 46%]
test/muppy/test_mprofile.py .                                            [ 47%]
test/muppy/test_muppy.py ...........                                     [ 56%]
test/muppy/test_refbrowser.py .....                                      [ 61%]
test/muppy/test_summary.py ........                                      [ 67%]
test/muppy/test_tracker.py ...                                           [ 70%]
test/test_process.py .....                                               [ 74%]
test/tracker/test_classtracker.py ....................                   [ 91%]
test/tracker/test_stats.py ..........                                    [100%]

=================================== FAILURES ===================================
________________________ GarbageTestCase.test_edges_new ________________________

self = <test.gui.test_garbage.GarbageTestCase testMethod=test_edges_new>

    def test_edges_new(self):
        """Test referent identification for new-style classes.
        """
        foo = FooNew()
        bar = BarNew()
    
        idfoo = id(foo)
        idfd = id(foo.__dict__)
        idbar = id(bar)
        idbd = id(bar.__dict__)
    
        foo.next = bar
        bar.prev = foo
    
        del foo
        del bar
    
        gb = GarbageGraph()
    
>       self.assertTrue(_Edge(idfoo, idfd, '__dict__') in gb.edges, gb.edges)
E       AssertionError: False is not true : {<7ffff6880440 => 7ffff5936e40, '', None>, <7ffff5936e40 => 7ffff6880440, '', None>}

test/gui/test_garbage.py:202: AssertionError
________________________ GarbageTestCase.test_edges_old ________________________

self = <test.gui.test_garbage.GarbageTestCase testMethod=test_edges_old>

    def test_edges_old(self):
        """Test referent identification for old-style classes.
        """
        foo = Foo()
        bar = Bar()
    
        idfoo = id(foo)
        idfd = id(foo.__dict__)
        idbar = id(bar)
        idbd = id(bar.__dict__)
    
        foo.next = bar
        bar.prev = foo
    
        del foo
        del bar
    
        gb = GarbageGraph()
    
>       self.assertTrue(_Edge(idfoo, idfd, '__dict__') in gb.edges, gb.edges)
E       AssertionError: False is not true : {<7ffff6880440 => 7ffff5936e40, '', None>, <7ffff5936e40 => 7ffff6880440, '', None>}

test/gui/test_garbage.py:178: AssertionError
__________________________ GarbageTestCase.test_split __________________________

self = <test.gui.test_garbage.GarbageTestCase testMethod=test_split>

    def test_split(self):
        """Test splitting into subgraphs.
        """
        foo = Foo()
        bar = Bar()
    
        idfoo = id(foo)
        idbar = id(bar)
        idfd = id(foo.__dict__)
        idbd = id(bar.__dict__)
    
        foo.next = bar
        bar.prev = foo
    
        l = []
        l.append(l)
        idl = id(l)
    
        del foo
        del bar
        del l
    
        gb = GarbageGraph()
        subs = gb.split_and_sort()
        self.assertEqual(len(subs), 2)
    
>       self.assertEqual(subs[0].count, 4)
E       AssertionError: 2 != 4

test/gui/test_garbage.py:116: AssertionError
=============================== warnings summary ===============================
test/asizeof/test_asizeof.py: 21100 warnings
test/tracker/test_classtracker.py: 10587 warnings
  /build/pympler-1.1/pympler/asizeof.py:368: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
    if a.startswith(pref) and hasattr(obj, a) and a not in excl:

test/asizeof/test_asizeof.py: 21100 warnings
test/tracker/test_classtracker.py: 10587 warnings
  /build/pympler-1.1/pympler/asizeof.py:369: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
    yield a, getattr(obj, a)

test/tracker/test_classtracker.py::SnapshotTestCase::test_background_monitoring
  /build/pympler-1.1/pympler/classtracker.py:513: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
    self._periodic_thread.setDaemon(True)

test/tracker/test_classtracker.py::SnapshotTestCase::test_background_monitoring
  /build/pympler-1.1/test/tracker/test_classtracker.py:231: DeprecationWarning: getName() is deprecated, get the name attribute instead
    self.assertEqual(self.tracker._periodic_thread.getName(), 'BackgroundMonitor')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test/gui/test_garbage.py::GarbageTestCase::test_edges_new - AssertionError: False is not true : {<7ffff6880440 => 7ffff5936e40, '', Non...
FAILED test/gui/test_garbage.py::GarbageTestCase::test_edges_old - AssertionError: False is not true : {<7ffff6880440 => 7ffff5936e40, '', Non...
FAILED test/gui/test_garbage.py::GarbageTestCase::test_split - AssertionError: 2 != 4
========= 3 failed, 115 passed, 9 deselected, 63376 warnings in 32.05s =========


```

</details>

## Additional context

Discovered by @gregsgit

## Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.7-arch1-1, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/home/graham/.nix-defexpr/channels/nixpkgs`
```

## Notify maintainers

@mweinelt 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pympler/default.nix
+++ b/pkgs/development/python-modules/pympler/default.nix
@@ -45,6 +45,12 @@ buildPythonPackage rec {
       "test_findgarbage"
       "test_get_tree"
       "test_prune"
+    ]
+    ++ lib.optionals (pythonAtLeast "3.13") [
+      # https://github.com/pympler/pympler/issues/163
+      "test_edges_new"
+      "test_edges_old"
+      "test_split"
     ];
 
   doCheck = stdenv.hostPlatform.isLinux;

